algorithms/ar-msrt/typescript/merge_sort.ts

```diff
@@ -0,0 +1,30 @@
+function mergeSort(array: number[]) : number[]{
+    const merge = function (array1 : number[], array2 : number[]) : number[] {
```

Comment:
Codacy found an issue: [Function has a complexity of 10. Maximum allowed is 5.](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6334421)